src/AutoRest.CSharp/MgmtTest/Generation/Mock/MgmtMockTestBaseWriter.cs

```diff
@@ -44,6 +48,75 @@ public virtual void Write()
             }
         }
 
+        public virtual void WriteSample(MockTestCase testCase)
+        {
+            generatingSample = true;
+            Regex pattern = new Regex(@".*(Azure.ResourceManager.\w+)/.*");
+            Match match = pattern.Match(Configuration.OutputFolder);
+            if (match.Success)
+            {
+                _writer.additionalComments.Add($"// Need to install {match.Groups[1].Value} v1.0.0 to execute below sample code");
+            }
+            _writer.addUsingRegion = true;
+            _writer.UseNamespace("Azure.ResourceManager");
+            _writer.UseNamespace("System");
+            _writer.UseNamespace("Azure.Identity");
+            _writer.UseNamespace("System.Threading.Tasks");
+
+            _writer.Line($"#region Parameter decalarations");
```

Comment:
Maybe we could consider to add a new method in codeWriter to make sure the code writer to write regions (like CodeWriter.Scope). This could make sure we will never forget to append the `#endregion`